wire/error.go

```diff
@@ -1,14 +1,138 @@
+	// the maximum allowed.
+	ErrTooManyLocators
+	// ErrTooManyVectors is returned when the number of inventory vectors
+	// exceed the maximum allowed.
+	ErrTooManyVectors
+	// ErrTooManyHeaders is returned when the number of block headers exceed
+	// the maximum allowed.
+	ErrTooManyHeaders
+	// ErrHeaderContainsTxs is returned when a header's transactions
+	// count is greater than zero.
+	ErrHeaderContainsTxs
+	// ErrTooManyVotes is returned when the number of vote hashes exceed the
+	// maximum allowed.
+	ErrTooManyVotes
+	// ErrTooManyBlocks is returned when the number of block hashes exceed the
+	// maximum allowed.
+	ErrTooManyBlocks
+	// ErrUnequalQuantities is returned when transactions quantities expected
+	// to be equal are not.
+	ErrUnequalQuantities
+	// ErrUnknownTxType is returned when a transaction type is unknown.
+	ErrUnknownTxType
+	// ErrWitnessOnlyTx is returned when reading the prefix for a witness only
+	// transaction.
+	ErrWitnessOnlyTx
+	// ErrInvalidMsg is returned when for an invalid  message structure.
+	ErrInvalidMsg
+	// ErrUserAgentTooLong is returned when the provided user agent exceeds
+	// the maximum allowed.
+	ErrUserAgentTooLong
+	// ErrTooManyFilterHeaders is returned when the number of committed filter
+	// headers exceed the maximum allowed.
+	ErrTooManyFilterHeaders
+	// numErrorCodes is the maximum error code number used in tests.  This
+	// entry MUST be the last entry in the enum.
+	numErrorCodes
 )
 
+// Map of ErrorCode values back to their constant names for pretty printing.
+var errorCodeStrings = map[ErrorCode]string{
+	ErrNonCanonicalVarInt:   "ErrNonCanonicalVarInt",
+	ErrVarStringTooLong:     "ErrVarStringTooLong",
+	ErrVarBytesTooLong:      "ErrVarBytesTooLong",
+	ErrCmdTooLong:           "ErrCmdTooLong",
+	ErrPayloadTooLarge:      "ErrPayloadTooLarge",
+	ErrWrongNetwork:         "ErrWrongNetwork",
+	ErrMalformedCmd:         "ErrMalformedCmd",
+	ErrUnknownCmd:           "ErrUnknownCmd",
+	ErrPayloadChecksum:      "ErrPayloadChecksum",
+	ErrTooManyAddrs:         "ErrTooManyAddrs",
+	ErrTooManyTxs:           "ErrTooManyTxs",
+	ErrInvalidVersion:       "ErrInvalidVersion",
+	ErrFilterTooLarge:       "ErrFilterTooLarge",
+	ErrTooManyProofs:        "ErrTooManyProofs",
+	ErrTooManyFilterTypes:   "ErrTooManyFilterTypes",
+	ErrTooManyLocators:      "ErrTooManyLocators",
+	ErrTooManyVectors:       "ErrTooManyVectors",
+	ErrTooManyHeaders:       "ErrTooManyHeaders",
+	ErrHeaderContainsTxs:    "ErrHeaderContainsTxs",
+	ErrTooManyVotes:         "ErrTooManyVotes",
+	ErrTooManyBlocks:        "ErrTooManyBlocks",
+	ErrUnequalQuantities:    "ErrUnequalQuantities",
+	ErrUnknownTxType:        "ErrUnknownTxType",
+	ErrWitnessOnlyTx:        "ErrWitnessOnlyTx",
+	ErrInvalidMsg:           "ErrInvalidMsg",
+	ErrUserAgentTooLong:     "ErrUserAgentTooLong",
+	ErrTooManyFilterHeaders: "ErrTooManyFilterHeaders",
+}
+
+// String returns the ErrorCode as a human-readable name.
+func (e ErrorCode) String() string {
```

Comment:
Personally, I'd make `ErrorCode` implement the `error` interface and then define the `Is()` function for it. That would make it play better with the rest of the go1.13 error ecosystem as you could simply write:

`if errors.Is(err, wire.ErrTooManyProofs) { ... }`